pkg/core/function.go

```diff
@@ -120,18 +115,27 @@ func (c *client) CreateFunction(buildpackBuilder Builder, options CreateFunction
 		}
 	} else {
 		// buildpack based cluster build
+		spec := &build.BuildSpec{
+			ServiceAccountName: "riff-build",
+			Source:	c.makeBuildSourceSpec(options),
+			Template: &build.TemplateInstantiationSpec{
+				Name: "riff-cnb",
+				Arguments: c.makeBuildArguments(options),
+			},
+		}
 		s.Spec.RunLatest.Configuration.Build = &v1alpha1.RawExtension{
-			BuildSpec: &build.BuildSpec{
-				ServiceAccountName: "riff-build",
-				Source:             c.makeBuildSourceSpec(options),
-				Template: &build.TemplateInstantiationSpec{
-					Name:      "riff-cnb",
-					Arguments: c.makeBuildArguments(options),
+			Object: &unstructured.Unstructured{
```

Comment:
it shouldn't be necessary to fallback to the unstructured option since we are using a Build resource.